src/status_im2/contexts/chat/bottom_sheet_composer/actions/view.cljs

```diff
@@ -0,0 +1,143 @@
+(ns status-im2.contexts.chat.bottom-sheet-composer.actions.view
+  (:require
+    [quo2.core :as quo]
+    [react-native.core :as rn]
+    [react-native.permissions :as permissions]
+    [react-native.platform :as platform]
+    [react-native.reanimated :as reanimated]
+    [reagent.core :as reagent]
+    [status-im2.common.alert.events :as alert]
+    [status-im2.contexts.chat.bottom-sheet-composer.constants :as c]
+    [status-im2.contexts.chat.messages.list.view :as messages.list]
+    [utils.i18n :as i18n]
+    [utils.re-frame :as rf]
+    [status-im2.contexts.chat.bottom-sheet-composer.actions.style :as style]))
+
+(defn send-message
+  [{:keys [input-ref]}
+   {:keys [text-value focused?]}
+   {:keys [height saved-height last-height opacity background-y container-opacity]}
+   window-height]
+  (reanimated/animate height c/input-height)
+  (reanimated/set-shared-value saved-height c/input-height)
+  (reanimated/set-shared-value last-height c/input-height)
+  (reanimated/animate opacity 0)
+  (when-not @focused?
+    (js/setTimeout #(reanimated/animate container-opacity c/empty-opacity) 300))
+  (js/setTimeout #(reanimated/set-shared-value background-y
+                                               (- window-height))
+                 300)
+  (rf/dispatch [:chat.ui/send-current-message])
+  (rf/dispatch [:chat.ui/set-input-maximized false])
+  (rf/dispatch [:chat.ui/set-input-content-height c/input-height])
+  (reset! text-value "")
+  (when @input-ref
+    (.clear ^js @input-ref))
+  (messages.list/scroll-to-bottom))
+
+(defn send-button
+  [props
+   {:keys [text-value] :as state}
+   animations
+   window-height
+   images?]
+  [:f>
```

Comment:
just wondering why two `:f>` components are needed here ?

Reply:
So that `btn-opacity` and `z-index` do not overwrite values on re-render